### DISCUSSION TITLE
Bevy high CPU load on simple examples

### DISCUSSION THREAD

**Discussion by fbreuer** [bevyengine/bevy#6930] [Q&A]
I am seeing high CPU load across Bevy examples. For example, on the sprite.rs example, I get 130% CPU and 30% GPU load on Mac OS X 13.0.1, MacBook 2017, 1.3 GHz Dual-core i5, Intel HD Graphics 615. Granted, this is an old system, but given this example is just displaying a static image, this load still seems excessive. I have seen various issues (https://github.com/bevyengine/bevy/issues/4718, https://github.com/bevyengine/bevy/issues/5713) and PRs (https://github.com/bevyengine/bevy/pull/6503) discussing CPU load, but not being familiar with the details, I am not sure what the upshot of all of this is. So my questions:

For something simple like sprite.rs, what load should I expect to see (on Mac OS X)?

If the answer is "a lot", is this reasonably expected to change any time in the foreseeable future?

**Comment by torsteingrindvik** [comment]
Are you running with `--release`?

EDIT: 
Here a simple snapshot of usage for me via task manager:

Debug:
![bilde](https://user-images.githubusercontent.com/52322338/207080605-ab72239c-4c64-4b2a-a9c3-132eb1c3cb85.png)

Release:
![bilde](https://user-images.githubusercontent.com/52322338/207081089-16c736ee-bd9d-4945-837f-e06a3cc43b4e.png)

Although I'm currently on Win10, 2080ti, Ryzen 9 3900x. Different HW than yours but I still think your numbers sound high.

**Reply by tim-blackbird** [reply]
I'll add to this that cargo's default dev profile is not made with gamedev in mind.
Bevy recommends you enable some optimizations for the dev profile [here](https://bevyengine.org/learn/book/getting-started/setup/#compile-with-performance-optimizations)

**Reply by fbreuer** [reply]
Thanks! This is very useful context.

I rebuilt with --release. This improved by a factor of 2, getting me down to about 55% CPU and 45% GPU.

This is better, but still high. I will if I can wrap my head around profiling to get a better understanding of what is going on.
